packages/lesswrong/server/resolvers/tagResolvers.ts

```diff
@@ -17,6 +19,99 @@ import take from 'lodash/take';
 import filter from 'lodash/filter';
 import * as _ from 'underscore';
 import { recordSubforumView } from '../../lib/collections/userTagRels/helpers';
+import {
+  defaultSubforumSorting,
+  SubforumSorting,
+  subforumSortings,
+  subforumSortingToResolverName,
+  subforumSortingTypes,
+} from '../../lib/subforumSortings';
+
+type SubforumFeedSort = {
+  posts: SubquerySortField<DbPost, keyof DbPost>,
+  comments: SubquerySortField<DbComment, keyof DbComment>,
+  sortDirection?: SortDirection,
+}
+
+const subforumFeedSortings: Record<SubforumSorting, SubforumFeedSort> = {
+  magic: {
+    posts: { sortField: "score" },
+    comments: { sortField: "score" },
+  },
+  new: {
+    posts: { sortField: "postedAt" },
+    comments: { sortField: "postedAt" },
+  },
+  old: {
+    posts: { sortField: "postedAt", sortDirection: "asc" },
+    comments: { sortField: "postedAt", sortDirection: "asc" },
+    sortDirection: "asc",
+  },
+  top: {
+    posts: { sortField: "baseScore" },
+    comments: { sortField: "baseScore" },
+  },
+  recentComments: {
+    posts: { sortField: "lastCommentedAt" },
+    comments: { sortField: "lastSubthreadActivity" },
+  },
+}
+
+const createSubforumFeedResolver = <SortKeyType>(sorting: SubforumFeedSort) => async ({
+  limit = 20, cutoff, offset, args: {tagId, af}, context,
+}: {
+  limit?: number,
+  cutoff?: SortKeyType,
+  offset?: number,
+  args: {tagId: string, af?: boolean},
+  context: ResolverContext,
+}) => mergeFeedQueries({
+  limit,
+  cutoff,
+  offset,
+  sortDirection: sorting.sortDirection,
+  subqueries: [
+    // Subforum posts
+    viewBasedSubquery({
+      type: "tagSubforumPosts",
+      collection: Posts,
+      ...sorting.posts,
+      context,
+      selector: {
+        [`tagRelevance.${tagId}`]: {$gte: 1},
+        hiddenRelatedQuestion: undefined,
+        shortform: undefined,
+        groupId: undefined,
+        ...(af ? {af: true} : undefined),
+      },
+    }),
+    // Subforum comments
+    viewBasedSubquery({
+      type: "tagSubforumComments",
+      collection: Comments,
+      ...sorting.comments,
+      context,
+      selector: {
+        tagId,
```

Comment:
```suggestion
        tagId,
        tagCommentType: "SUBFORUM",
        topLevelCommentId: {$exists: false},
```
To get only top level subforum comments (the replies are included in the top level comments when they are resolved)